### PR TITLE
[EDA-1529] board class  created & place_character_initial_pos method moved

### PR DESCRIPTION
**Pull request by lgiordano-eb**
- Board class created
- place_character_initial_pos was moved to the board class.
it method was modified because with the new class diagram the board class is responsible for placing the characters that the player owns. 

- tests were modified to evaluate if the character is in the correct place and if the cell where the character was placed changed the is_discover status

### CODE REVIEW

File: game/board.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from game.cell import Cell
+from constans.constants_game import (
+    LARGE
+)
```
Comment (ibrizuela-eb) [inline game/board.py:4]: nit: Alphabetical order and one line for single import
```suggestion
from constans.constants_game import LARGE
from game.cell import Cell
```

File: game/board.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class Board():
+    def __init__(self) -> None:
+        self._board = [
+            [Cell(i, j) for j in range(LARGE)] for i in range(LARGE)
```
Comment (ibrizuela-eb) [inline game/board.py:10]: nit: It has more sense if you give names to variables. (Please check if the names orders are okay)
```suggestion
            [Cell(row, col) for col in range(LARGE)] for row in range(LARGE)
```

File: game/board.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        ]
+
+    def place_character_initial_pos(
+        self,
+        characters: list,
```
Comment (ibrizuela-eb) [inline game/board.py:15]: list of what? str, int, tuples,...

File: game/game.py [hunk 4]
```diff
@@ -47,9 +47,14 @@
 class WumpusGame():
 
     def __init__(self) -> None:
+
+        # --------------------------------
```
Comment (ibrizuela-eb) [inline game/game.py:51]: nit: this kind of comments aren't used in PRODUCTION environments (at Eventbrite)
```suggestion
```

File: game/game.py [hunk 5]
```diff
@@ -47,9 +47,14 @@
...
         self._board = [
             [Cell(i, j) for j in range(LARGE)] for i in range(LARGE)
         ]
+
+        # --------------------------------
+
```
Comment (ibrizuela-eb) [inline game/game.py:57]: nit: this kind of comments aren't used in PRODUCTION environments (at Eventbrite)
```suggestion
```

File: test/test_board.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import unittest
+from constans.constans import (
+    PLAYER_1,
+    PLAYER_2,
+    INITIAL_POSITION_PLAYER_1,
+    INITIAL_POSITION_PLAYER_2
+    )
```
Comment (ibrizuela-eb) [inline test/test_board.py:7]:
```suggestion
)
```

File: test/test_board.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
+import unittest
+from constans.constans import (
+    PLAYER_1,
+    PLAYER_2,
+    INITIAL_POSITION_PLAYER_1,
+    INITIAL_POSITION_PLAYER_2
```
Comment (ibrizuela-eb) [inline test/test_board.py:6]: missing trailing comma at the end of `INITIAL_POSITION_PLAYER_2` and not in alphabetical order
```suggestion
    INITIAL_POSITION_PLAYER_1,
    INITIAL_POSITION_PLAYER_2,
    PLAYER_1,
    PLAYER_2,
```

File: test/test_board.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    )
+from game.board import Board
+from game.player import Player
+
+from constans.constants_game import (
+    LARGE,
+)
```
Comment (ibrizuela-eb) [inline test/test_board.py:13]: - Remove blank space
- Wrong order
- One line import
```suggestion
from constans.constants_game import LARGE
from game.board import Board
from game.player import Player
```

File: test/test_board.py [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    def test_place_character_initial_pos_player_1(self):
+        board = Board()
+        player_1 = Player(PLAYER_1)
+        board.place_character_initial_pos(
+            player_1.characters,
+            INITIAL_POSITION_PLAYER_1,
+            0)
+        for index, character in enumerate(player_1.characters):
+            row, col = INITIAL_POSITION_PLAYER_1[index]
+            self.assertEqual(board._board[row][col].character, character)
```
Comment (ibrizuela-eb) [inline test/test_board.py:32]: Using `for` loops in tests is not a good practice. We can discuss this later.

File: game/game.py (hunk 4 above)
Comment (lgiordano-eb) [reply game/game.py:51]: big mistake to leave these comments

File: test/test_board.py (hunk 9 above)
Comment (lgiordano-eb) [reply test/test_board.py:32]: Ok, I think that in those cases I could test by a parameterized instead in a for loop

File: game/board.py (hunk 1 above)
Comment (lgiordano-eb) [reply game/board.py:4]: Ok, I will correct all the imports

**Review by mmerida-eb — APPROVED** [review]

**Pull request merged**